pandas/tests/io/formats/test_style.py

```diff
@@ -1173,6 +1173,25 @@ def test_hide_columns_mult_levels(self):
         assert ctx['body'][1][2]['is_visible']
         assert ctx['body'][1][2]['display_value'] == 3
 
+    def test_pipe(self):
+        def set_caption_from_template(styler, a, b):
+            return styler.set_caption(
+                'Dataframe with a = {a} and b = {b}'.format(a=a, b=b))
+        styler = self.df.style.pipe(set_caption_from_template, 'A', b='B')
+        assert 'Dataframe with a = A and b = B' in styler.render()
+
+        def f(s, *args, **kwargs):
+            return s, args, kwargs
+
+        result = self.df.style.pipe(f, 0, a=1)
+        assert result[1] == (0,)
+        assert result[2] == dict(a=1)
+
+        def g(**kwargs):
+            assert 'styler' in kwargs
+            return kwargs['styler'].data
+        assert self.df.style.pipe((g, 'styler')) is self.df
```

Comment:
Can you comment here explaining why you're using `is` instead of `assert_frame_equal` ?

Reply:
Also, add a comment explaining why you're creating this `g` function in the first place (perhaps a more descriptive function name is needed).